packages/actor-init-sparql/lib/HttpServiceSparqlEndpoint.ts

```diff
@@ -158,8 +159,31 @@ Options:
   public async handleRequest(engine: ActorInitSparql, variants: { type: string, quality: number }[],
                              stdout: Writable, stderr: Writable,
                              request: http.IncomingMessage, response: http.ServerResponse) {
-    const mediaType: string = request.headers.accept && request.headers.accept !== '*/*'
-      ? require('negotiate').choose(variants, request)[0].type : null;
+
+    // Negotiate the best mediatype format
+    let mediaType = null;
+    const isValid = request.headers.accept && request.headers.accept !== '*/*';
+    let negotiation = isValid ? require('negotiate').choose(variants, request) : null;
+    mediaType = negotiation && negotiation.length > 0 ? negotiation[0].type : null;
+
+    if (negotiation && negotiation.length > 1) {
```

Comment:
I still think this can be simplified significantly.

Do you think something like this would work?

```
// Assuming variants is sorted by q in decreasing order (pseudocode)
// Create negotiationsIndexed so that it is a hash containing the media types from 'negotiation' as keys.
for (variant of variants) {
  if (negotiationsIndexed[variant]) {
    mediaType = variant;
    break;
  }
}
```